contracts/most/UtBase.sol

```diff
@@ -0,0 +1,228 @@
+pragma solidity >=0.5.0 <0.6.0;
+
+// Copyright 2020 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import "./GenesisUtBase.sol";
+
+import "../consensus/CoconsensusModule.sol";
+import "../proxies/MasterCopyNonUpgradable.sol";
+import "../utility-token/ERC20Token.sol";
+
+/**
+ * @title UtBase is an ERC20 token wrapper for the base coin that is used to
+ *        pay for gas consumption on the auxiliary chain.
+ */
+contract UtBase is MasterCopyNonUpgradable, GenesisUtBase, ERC20Token, CoconsensusModule {
+
+    /* Constants */
+
+    /** Token Symbol */
+    string public constant TOKEN_SYMBOL = "UB";
+
+    /** Token Name */
+    string public constant TOKEN_NAME = "UtBase";
+
+    /** Token Decimal */
+    uint8 public constant TOKEN_DECIMALS = 18;
+
+
+    /* Storage */
+
+    /** Consensus cogateway address defined in genesis. */
+    address private consensusCogateway = address(0x0000000000000000000000000000000000004d02);
+
+
+    /* Modifiers */
+
+    modifier onlyConsensusCogateway()
+    {
+        require(
+            msg.sender == getConsensusCogateway(),
+            "Caller must be consensus cogateway contract address."
+        );
+        _;
+    }
+
+
+    /* External Functions */
+
+    /**
+     * @notice Initializes the UtBase contract.
+     *
+     * @dev This function must be called only once.
+     *
+     *      Function requires:
+     *          - msg.sender should be Coconsensus
+     *          - setup can be called only once
```

Comment:
As per the dev guidelines, Please use `\pre` and `\post` conditions. 🤺

Reply:
Done.